### ISSUE TITLE
No longer compatible with IntelliJ 2021.2 & JB marked as deprecated

### ISSUE THREAD

**Issue by Tecktron** [vsch/idea-multimarkdown#918]
I can no longer find this in the marketplace on version 2021.x, are there plans to make this compatible?

I also noticed that on the JB site it's now listed as deprecated, is that true?
![image](https://user-images.githubusercontent.com/4650904/130001741-e9520142-fef2-4f44-82be-47b7f7aa4b19.png)

**Comment by vsch** [comment]
At the start of the year I took on a 4 to 6 week project which morphed into 6 month, 7 days a week black hole which made it impossible for me to chase the JetBrains API changes with every IDE update. 

Four months ago I was hoping to get a break of a couple of weeks to address compatibility issues with new releases of the IDE but that never materialized. JetBrains marked it as deprecated since the compatibility issues have not been addressed.

It now looks like I will now get a week between projects and I am hoping this will be enough to address compatibility with 2021.2. 

However, some issues are not addressable on my end because JetBrains is integrating their bundled markdown plugin into other plugins. There is nothing I can do to address this. Their markdown plugin and Markdown Navigator are mutually exclusive.

This started with Jupiter notes for python in 2019. At the time I contacted JetBrains and offered to do everything needed to allow Markdown Navigator to support whatever they needed. I understood that integration with their bundled markdown plugin is easier than creating a generic API for markdown parsing. Having received no reply I got the message that there was no interest on their end in making it happen and that in the long run Markdown Navigator will be pushed out by further integration with their bundled plugin.

This is something I knew would be a likely outcome when I wrote the plugin 6 years ago. However, there was no halfway decent markdown plugin so I dove in. It has been an economically unjustified labour of love for over five years.

To make a long story short, I am planning to release a compatibility update to address API changes in the new version but compatibility issues caused by conflict with the bundled markdown plugin are out of my control. I will probably have to remove PyCharm IDE from the compatibility list for Markdown Navigator.

Some users reported that bundled GitHub integration plugin also uses the bundled markdown plugin. If this is the case then there will be no point in updating Markdown Navigator for new releases because GitHub integration is an absolute must for the IDE.

In the long run the plugin is doomed because evolution of JetBrains IDEs is pushing it out of existence. It was my baby for over 5 years with me spending 7 days a week of 12-18 hour days on it with no vacations or breaks. So I do have mixed emotions about it. However, life goes on and this was an exhilarating ride that looks like it is probably coming to an end.

I will know better in a week when I will have had time to look into the issues in detail and decide if there is a way to extend the plugin's life.

**Comment by rayk** [comment]
This is sad because the markdown navigator has been a very useful tool, so much so with the knowledge it is not going to be around for the long run I still renewed my licence recently. Then I found out it would not run in 2021.2

@vsch I hope you get a break and to at least get running with 2021.

That said what is least impressive, is JetBrains not looking for a way to markdown navigators into their own offering, which is still very basic. This is not surprising I have seen the rise and fall of a few IDE's now, and there is a reoccurring pattern... I really hope that will not have to move to VsCode in the future... 


Good luck and looking forward to any updates.

**Comment by pgnd** [comment]
Anyone else completely fed up with JB's habit of limiting/breaking existing great 3rd party functionality, with great support, like @vsch 's  ?  And when they 'do' their own/native implementation, it's half-baked/half-assed at best?

The irony here is that over the past week, several folks here have received this message from JB:
```
We hope you've enjoyed using JetBrains products! We’d kindly like to inform you that your subscription(s) has expired on ..., 2021. If you'd like to keep your subscription(s) active, please renew online manually to avoid any interruptions.

The subscription(s) requiring payment:
1 × Markdown Navigator Enhanced

Please note that expired subscription(s) can only be renewed later through your JetBrains Account.

Our goal is to provide the best development tools that our customers enjoy using year after year. We appreciate any feedback about your experience.
```

for me, @vsch 's MD was one strong reason in favor of sticking with JB prods, despite their challenges with adequate modern support.
without it, the scales tip.  probably in favor of VSCode.

TL;DR thx @vsch

**Comment by fulopattila122** [comment]
@vsch I don't know if others would agree (thumbs up or thumbs down) but I recommend increasing the price of the plugin. I'm paying for much less complex plugins 5-10 euros a month. 5-15 EUR won't be an unusual price, and there's no benefit for me paying 1 EUR and then losing the plugin.

I don't know how many users you have, but do the math, and calculate what is the price that allows you to reserve time for maintenance. I don't think many new features are needed, Markdown Navigator is already packed and mature.

Maybe you could also send out an email to the users of your plugin, tell them about the situation, as you did above, and do a simple pricing survey. Likely the [Van Westendorp method](https://www.qualtrics.com/uk/experience-management/product/pricing-surveys) would be the best match, where you ask:

> 1. The price at which the product is so cheap that they would question its quality
> 2. The price at which the product is inexpensive, but not so inexpensive that they would question its quality.
> 3. The price at which the product is expensive, but not so expensive that they would still consider it
> 4. The price at which the product is so expensive that they would not consider it

IMHO Markdown Navigator is a great product, and letting it sink would be a big loss, especially for you, since you have invested 1000s of hours into it.

Anyway, whatever you decide will be accepted :smiley: :bow:

**Comment by qwazer** [comment]
HI @vsch 

It a good news that you appeared online :) I was worried about you.

I'm fan of you MD plugin and in case Jetbrains will not support it anymore i'm ready to use it as standalone md edition tool. Now I use your plugin on day-to-day basic. Besides usual project docs, I use it to write long Jira comments, long messages to chat/mails. And I still hope to find enough time and energy to reuse your excellent flexmark-java as I wrote in https://github.com/vsch/flexmark-java/issues/258.

Do you have any public JB issue to vote for it and leave a couple of comments from our user side?

Regards.

**Comment by pgnd** [comment]
> in case Jetbrains will not support it anymore

sales@JB just a couple of days ago responded to one of my customers that their "Plugin Marketplace team" is "concerned that the plugin was not being updated with our releases" & that they couldn't "reach the developer", so they "suspend(ed) sales" to protect their customers from a "product that was no longer being updated".

And that once "outstanding issues" are addressed they "can allow for the sales again."

I saw the entire email thread; personally, I don't believe their story.

> public JB issue to vote for it

ime, voting @ JB is generally useless.  see for example the @redis support 'voting' -- #1 issue, @ 1300 votes, 3 yrs of non-progress from JB.

the decision about this plugin lies here, with vsch.

**Comment by hsz** [comment]
@pgnd We’re constantly improving the quality of JetBrains Marketplace, support, and actual plugins delivered to end-users.
Since third-party plugins aren’t our property, we can’t do much about their issues.
As you’ve mentioned, the only way to protect folks from getting buried with exceptions is to mark plugin updates as incompatible (especially when it comes to paid plugins) and get in touch with authors to offer a guide on how to deal with specific problems.
In this particular case, we reached @vsch about the issues (in numbers: over 150 problems reported by the Plugin Verifier tool) – he replied about the planned update, but all that has happened in April.
Before any action, we get in touch with vendors – and now it was no different.

**Comment by pgnd** [comment]
> Before any action, we get in touch with vendors – and now it was no different.

You might want to get your story 'in-sync' wth that from your sales team.

**Comment by hsz** [comment]
@pgnd I'm aware of the thread you've mentioned, and still, my answer is valid.

Since April, we didn't get any confirmation from the plugin author or any feedback about the plugin deprecation and planned updates.
As I said, we're happy to help, but after five months of inactivity, we had to take proper steps. 

And don't get me wrong, I don't blame Vladimir here, but paid plugins should be properly maintained when it comes to money. In such a case, JetBrains is the first who'll be accused of delivering incompatible features, not the third-party vendor.

There are many ways of providing the help we offer for plugin developers, and AFAIK, @vsch is aware of all of them.

**Comment by pgnd** [comment]
> my answer is valid.

call it whatever you like.

it doesn't change the facts that
(a) that's NOT what your sales person told the customer, referencing communications from your "PLugin Marketplace Team"
(b) JB failed to communicate that there was any issue to the customer; in fact, it communicated the OPPOSITE -- that the app WAS available for renewal, and WAS going to be billed.  From the screenshots I've seen, it was still in the Marketplace as available after you took the customer's renewal fee.  The ONLY indication that there was an issue was that you took their money for only the _other_ subscriptions on their renewal -- minus the MarkdownNavigator renewal.
(c) @vsch 's app -- regardless of whether it ulitmately ends up still being available for JB or not -- is a primary reason for some, myself included, for using JB.  without it, or a reasonable functional alternative (no, that's not the 'native' MD), it's a non starter.

**Comment by fulopattila122** [comment]
I can accept @hsz answer on that, @pgnd let's consider the other party as well, just put yourself in their shoes.

Nevertheless, I also had a very misleading experience, because my PhpStorm told me it was not possible to renew the subscription. I had the impression it's my mistake and did a couple of (useless) measures. So the way JB has communicated it towards subscribers, well.. there's definitely space for improvement :)

**Comment by qwazer** [comment]
My license is expired, cannot renew in JB marketplace. After reinstalling the plugin from disk it doesn't require a licence and all functionally available.

**Comment by fulopattila122** [comment]
@vsch Was your comment here a definitive "No it won't be upgraded ever" statement? I'm asking because if you made your decision, then - even if sadly - I'll acknowledge it and will step further and seek for alternatives.

So I'd appreciate having a black&white answer on that. Thx! :bow:

**Comment by bitbonk** [comment]
Oh no, this plugin was so super freaking cool, it had sooo many useful features that the bundled markdown plugin didn't have. It was a joy to use. It is quite painful having to work without it now. What can we do to bring it back?

**Comment by m-radzikowski** [comment]
Even more problems on WebStorm 2021.3:
![image](https://user-images.githubusercontent.com/4042673/144256770-da47fb80-933f-4adf-abb0-d124f5d096c1.png)

Plugin get's automatically disabled.

**Comment by alexeykuptsov** [comment]
There are 125 forks of this project in GitHub. Did anybody manage to make some progress in adapting this great plugin to new IDE versions?

**Comment by qwazer** [comment]
@alexeykuptsov. Based on https://github.com/vsch/idea-multimarkdown/network the most probable answer is "No".

**Comment by mvastola** [comment]
So for the first time, when I submit an error report I am now seeing this:
![image](https://user-images.githubusercontent.com/599767/148585800-7e0cc2ce-249b-475c-ba55-d4c2a450a9d2.png)

Does this mean it's being worked on again? (I don't see that version released anywhere, in which case I would guess it's upcoming?) Or has it been there for a while and I just haven't noticed?

**Comment by khun84** [comment]
@vsch Im a paid subscriber of this plugin. This plugin made my technical documentation work swift and easy and I couldn't imagine how unproductive it would be without it.

Really appreciate your effort put in to develop and maintain this plugin. Hope there's a way out that would make this work for all of us.

**Comment by qwazer** [comment]
new issue with intellij-idea-2022-1-eap-4 #927

**Comment by ogradyjd** [comment]
I have voiced my "disappointment" with Jetbrains on this matter.  
https://youtrack.jetbrains.com/issue/IDEA-266596

**Comment by vsch** [comment]
I have not commented on this issue, because I was not sure what to say and I did not want to lash out in emotionally charged anger. The decision to stop working on Markdown Navigator was an emotionally difficult one but my circumstances and JetBrains unstated attitude towards the MdNav plugin, made it impossible for me to continue plugging away as if the circumstances will one day improve.

This comment by **[hsz]**, specifically sent my blood boiling and I decided to let time do its thing. I did not know it would take almost two years:

> In this particular case, we reached @vsch about the issues (in numbers: over 150 problems reported by the Plugin Verifier tool) – he replied about the planned update, but all that has happened in April.

1. A release of the IDE introduced a bug into settings persistence that broke MdNav's settings [Settings style not saved · Issue #585 · vsch/idea-multimarkdown · GitHub]. I caught the bug early and reported it (<https://youtrack.jetbrains.com/issue/IDEA-184506>) in 2018.1.5 EAP. Two weeks later I posted a patch with a one-line fix. **It took months** to get that fixed in version 2018.2. For all the months before the fix, I would patch my IDE versions with a fix (copying affected java class files and rebuilding the affected jars, then signing the app on OSX), because the bug prevented me from being able to use my plugin.

2. The implementation of `editorActionHandler` extension, would report MdNav as the plugin at fault when other plugins caused an exception, because MdNav was the first in the extension list and passed the call to the original handler.

   This bug caused me to go through 780+ bug reports in 8 months, which were caused by other plugins and JetBrains IDE itself. I spent many days trying to mitigate the problem and reduce the watershed of bug reports.

3. In 2019, I was gunning to release MdNav for JetBrains Marketplace, in the hope that it would simplify purchasing and boost sales. The implementation of the marketplace made it impossible to have a single plugin with multiple types of licenses. I had to produce two licensed versions and figure out and implement an "easy" upgrade method for users. I had no interest in maintaining my own licensing scheme, which was only implemented because at the time JetBrains had no marketplace.

   This was a tangential effort, but I did manage to get over it. However, 2019.3-RC if memory serves correctly, changed the bundled Kotlin version which made constructors more strict. This broke an old Kotlin library of mine, written when Kotlin was 1.0 and which was used by MdNav and required a rewrite.

   I was tired of just doing stop-gap changes to make the plugin work with the latest release and took the time to update flexmark-java and MdNav with major, long planned rewrites. Five months later I was done.

   My beef is not that Kotlin was upgraded or that it broke the library. My beef that such a major backwards incompatible change was shoe-horned late in the release pipeline, giving no time for plugin developers to adjust. At the time, I started my day by updating intellij-community master branch, doing a build and using it for plugin development to ensure that upcoming releases did not break the plugin. After all that effort to be railroaded by a sudden change, was a rude awakening to realities of being a third party plugin developer for JetBrains IDEs.

4. The bundled Markdown plugin has taken a bite out of MdNav sales and only managed to beat it in downloads months after it was bundled and offered up on new installation. One user complained that MdNav needs to be more promoted because he wasted months working with JB Markdown, before discovering that MdNav was available. So JB Markdown is not really a competitor to MdNav, and I never viewed it as such. However, when it was made as a dependency in other JB plugins, I took it as a sign that JB really wanted MdNav out, without having the extremities to explicitly say so.

All in all, I can sympathize with JB Support for the user issues that MdNav has caused over the years, but many of these issues were caused by JetBrains **Quantum** API. While others were caused by the fact that I started on MdNav when the API docs were thin, systematic support for plugin developers was non-existent, and it took a long time to grasp the complexity of the API through trial and error. Part of the problem was that NdNav was popular:

![MdNavDownloads](https://user-images.githubusercontent.com/10299086/236691837-c3dac1d0-32b7-41e0-858a-d8238a482575.png)

To give a bit of history, which is always important when looking at current events, because it provides "the why" to "the what" happened.

For the first three years of development, I worked 16-hour days, 7 days a week. First year was all on the plugin and learning the API through trial and error. The next two years was taken up by writing [flexmark-java] library to replace the [pegdown] library, which was causing most of the plugin issues.

My hope was to reach a point when I could work on the plugin full time, which is a must for JetBrains plugins of meaningful complexity. However, this never panned out as JetBrains started bundling and promoting their Markdown Support plugin, now Markdown.

In the process of doing development, financially I was scraping bottom. The first two years I did not make $1/hr for my effort, the next couple of years boosted overall revenue to less than $5/hr. The situation only improved when I started consulting full-time and doing plugin maintenance in the second shift. This was not a solution, because I still worked double shifts and JetBrains API evolution caused me to spend most of my time updating the plugin for API changes, instead of plugin improvements.

It took three years to get over the emotional shock and try to re-evaluate the situation. I stayed away from my computer, not even turning it on. I ignored almost all e-mails, till my inbox glowed red with almost 10,000 unread e-mails. To say that I tuned it all out is an understatement. I doubted that I would ever try to work with JetBrains APIs.

Then two months ago, I had to complete an Arduino based project and wanted to use JetBrains CLion IDE with my Arduino Support plugin, because the Arduino IDE is a toy. It is notepad circa Windows95 glued to a build system.

Within hours of working with CLion, I had to get my must-have plugins updated, mainly Missing In Actions and Markdown Navigator. Arduino Support plugin too needed some attention. I wrote it five years ago, over the Christmas holidays, so I could have JetBrains IDE comforts, and never revisited meaningfully since.

I had to either work without Missing In Actions or Markdown Navigator or start working with JetBrains API, again.

I decided to give it another try. I was seriously hoping that the API has stabilized with the push to "dynamic everything" model started in 2018/2019. Like they say: "Hope springs eternal." but more appropriately: "Hope is a four-letter word."

I spent the last two months updating my plugins for gradle build system from PluginDevKit, update dependent libraries to be Maven accessible, and migrate away from deprecated API use.

I resisted gradle builds, originally, simply to avoid being a pioneer. I had enough arrows in my back by being an early Kotlin adopter. Not only was I right to avoid it early, I have mixed emotions about using it now. I find gradle builds solve complexity of dependencies, which is welcome. They also make build/debug cycle much longer and eliminate class reloading of application under debug. This means that there is an extra two to five minutes added to the cycle. Too long to be unnoticeable, too short for a cigarette break.

As the process advanced, I kept getting this feeling of deja-vu all over again. The docs are out of date, the API is still in a roiling boil, the IDEs are no more stable or meaningfully faster than when I walked away. Sure the IDE pops-up fast, but instead of taking a few minutes to come up with everything ready for user interaction, it now comes up immediately and the delays are smeared across tasks and features. I prefer the former. I could start the IDE and go for a coffee and a smoke. Come back and be productive. Now I have to sit there, sometimes waiting a minute for what I need to load and update.

Overall the development improvements are two steps forward, one step back, and sometimes, vice-versa.

I did get Markdown Navigator functional and resolved all relevant issue that the API churn permitted ([Deprecated EditorWindow.INITIAL_INDEX_KEY with replacement marked as ApiStatus
Internal], [ProjectPlainTextFileTypeManager.getFiles is deprecated replacement PersistentFileSetManager.getFiles is package private]).

Then I did what I used to do on a regular basis and build the intellij-community master, updated this morning and tried using it. I discovered another public API breaking change creeping into the code. I am almost certain that someone would catch this before it made it into a release, but as previous experience tells me, there are no guarantees.

What am I going to do? Good question to which I still have no definite answer. I wanted to restart my plugins, by moving some of them to paid, so I can justify the serious maintenance effort they take, and upping the price on Markdown Navigator as has been suggested. However, JetBrains API is moving faster with less stability. By releasing paid plugins, I would be making a commitment to making my plugins work with the quantum state of the API presented in future releases. I cannot say I can make this commitment lightly.

This morning, I found myself seriously entertaining the thought that I might have to find another development environment with which to work. I really love the IDEs, but I don't know for how long I can keep tolerating the unstable IDEs with constant API churn, UI evolution and icon evolution over stability, bug fixes and user requested features.

[Deprecated EditorWindow.INITIAL_INDEX_KEY with replacement marked as ApiStatus Internal]: https://intellij-support.jetbrains.com/hc/en-us/community/posts/11324174701970-Deprecated-EditorWindow-INITIAL-INDEX-KEY-with-replacement-marked-as-ApiStatus-Internal
[ProjectPlainTextFileTypeManager.getFiles is deprecated replacement PersistentFileSetManager.getFiles is package private]: https://intellij-support.jetbrains.com/hc/en-us/community/posts/11323711731346-ProjectPlainTextFileTypeManager-getFiles-is-deprecated-replacement-PersistentFileSetManager-getFiles-is-package-private
[Settings style not saved · Issue #585 · vsch/idea-multimarkdown · GitHub]: https://github.com/vsch/idea-multimarkdown/issues/585 "Settings style not saved · Issue #585 · vsch/idea-multimarkdown · GitHub"
[flexmark-java]: https://github.com/vsch/flexmark-java
[hsz]: https://github.com/hsz
[pegdown]: https://github.com/sirthias/pegdown

**Comment by vsch** [comment]
I spent the day mulling over the situation and options. No clarity to report. 

On my long deleted LinkedIn profile, I used to state that I have two levels of participation in everything I do:

1. Committed.
2. Not Involved.

So far, I could not find it in me to commit to maintaining my plugins without having my heart in it, and with everything I have experienced to date, I have not felt that change of heart. 

At the same time, I am not ready to abandon `Markdown Navigator Enhanced` nor `Missing In Actions`. I find these a staple in my development work. A few more people feel the same, judging from user feedback and comments.

I figured a temporary solution to buy time for attitudes and circumstances to change:

Release a version of Markdown Navigator Enhanced with compatibility fixes, where possible, to address IDEs from 2021.3 up to 2023.1.1, which I can verify work, since this is what I am running now. This release will be unlicensed and available to everyone. 

It will either be the last release or the last free release. In either case, users will have use of the plugin for another year or two, depending on how fast JetBrains API leaves the plugin behind and how fast users upgrade their IDEs.

It will buy me time to figure out whether I want to commit to supporting the plugin in the future as a paid plugin, or bite the bullet and give up on the effort as economically unsustainable. 

If I do decide to return it to the JetBrains Marketplace, it will also give me time to do some long overdue rewrites, or at least estimate the road map to have them implemented on a schedule.

Let me know if you find this compromise acceptable or if you have any other suggestions.

**Comment by fulopattila122** [comment]
Works, of course, and is acceptable. In fact, there's nothing unacceptable in _getting something for free_.

---

BUT. **How fair is it to you**?

I still suggest raising the price to $5-10/mo.
That's still a fair price, even just for buying the plugin "as-is" without support and new features.

Yes, there will be people moaning about it and a churn. But the churn has already happened.
I think you and your customers (including myself) must admit that it's impossible to maintain a product like this at a $1/mo price point.

Seeing the figures you shared, you'd likely make better money in some poor country as a factory worker.
No pun intended, but I don't see any point exploiting your work.

**You have a piece of excellent knowledge, ask the price it is worth**, please 

My 2 cents :bow:

**Comment by vsch** [comment]
Thank you for your support and understanding.

> Works, of course, and is acceptable. In fact, there's nothing unacceptable in _getting something for free_.
> 
> BUT. **How fair is it to you**?

All things considered, for me, it is just another drop in the ocean. It will allow supportive users who love and need the plugin to stop suffering for the impedance mismatch between me and JetBrains.
 
> I still suggest raising the price to $5-10/mo. That's still a fair price, even just for buying the plugin "as-is" without support and new features.
> 
> Yes, there will be people moaning about it and a churn. But the churn has already happened. I think you and your customers (including myself) must admit that it's impossible to maintain a product like this at a $1/mo price point.

I have come to grips that there are a lot of certifiable ultra-maroons (bugs-bunny speak), who think that software should be free, while they themselves have not produced anything at all or anything of value in the OSS space. I saw quite a few of these giving Md Nav one-star reviews because it is too expensive. One even had the nerve to tell me that it should be $1/yr. I told that one that he is sitting in front of a screen and keyboard. What's stopping him from doing it himself?

People don't realize how massive a task Markdown support in JetBrains IDEs really is. The IDE APIs are geared toward lexer and grammar based languages. Markdown does not fit that model. Every feature is a wrestling match with the API.

To put it in perspective, the plugin is 120k lines of Java/Kotlin code (comments and blank lines excluded), flexmark-java markdown library I wrote in support of the plugin is another 120k lines of java code, with 380k lines of Markdown, in tests and docs.

I would like to poll for volunteers willing to support this codebase for $5/hr, forget the effort it took to create it in the first place.

On the other hand, JetBrains boasts 16M users. If they were to offer the plugin bundled with the IDE for a $1/yr for personal and $5/yr corporate users, no one would even blink, much less complain, and I suspect a few million licenses would be sold. 

If they funnelled a small % of that revenue to me, added one or two dedicated in house resources to interface/work with me on the plugin. They would still have a few million $ of profit to add to their income statement. I did inquire about the possibility of bundling my plugin with the IDE license, about three or four years ago, but heard crickets instead of a response.
 
> Seeing the figures you shared, you'd likely make better money in some poor country as a factory worker. No pun intended, but I don't see any point exploiting your work.
>
> **You have a piece of excellent knowledge, ask the price it is worth**, please
> 
> My 2 cents 🙇

I agree with you on my ROI. It is pathetic. I can earn a lot more doing, literally, anything else. To give a bit of my background, I am going to be 60 this year, the only saving grace is that I look like I am 47. At 16, I started working as a full-time Z-80 Assembly language programmer, in 1979 for $980/mo. In 1986, at 23, I was consulting for a Hewlett-Packard division in Waterloo, Ontario, where I attended Waterloo University, Electrical Engineering. I was charging HP $1000/day + expenses, 7-days a week, for six months. I have charged similar rates since. In all cases, my clients made 10x more than I did in revenue or savings, and 5x more than me in profits, on my labour. No one paid me out of their pocket. I always made sure that my client's ROI on our relationship was at least 500%.

However, I do love text editing applications. Always did. I got hooked on programming at age 13 because I loved typing. I spent a lot of time at my mother's work in front of a DEC PDP-11 terminal. I felt stupid just typing garbage and needed something real to type, so I picked up a BASIC programming manual and wrote a 15-page program, using every feature the language offered. By the time I understood what I was doing by debugging it, I was hooked on programming for life.

I am mopping up old deprecated API use and will probably release 2021.3 to 2021.1.1 compatible version this week.

**Comment by ogradyjd** [comment]
Fully agree with @fulopattila122 

You've probably seen my comments in the Jetbrains forum about your plugin.  My work has definitely suffered without it.  Just let us know where we can still donate, because there is no way I'm going to use this in Comcast at our benefit for our hundreds of pages of documentation and not get you some kind of recompense. 

Your work is very, very appreciated.  It saved us hundreds of hours of time in having all our TOCs correct and making it easier for our internal users to find what they needed.  I can say that honestly because without MN, we've had to manually regenerate all our TOCs (when we remembered to) and the kindest thing I can say about Jetbrain's markdown plugin is that it exists.

They really should have hired you or bought your code from you for an appreciable amount.

**Comment by vsch** [comment]
@ogradyjd, thank you for your understanding. I read your comments and others. I even got one review in the e-mail, that got deleted in the JB Marketplace because the reviewer did not mince their words, and ripped JB a new.

In the end, it is all seems to be water off a duck. JB has no policy for third party plugin support. All decisions are handed off to the discretion of the developer in charge of the module: <https://intellij-support.jetbrains.com/hc/en-us/community/posts/11398505179666/comments/11408228756754>

I managed developers and can certify that some can be trusted completely to be presentable outside the company, others I would not allow interacting with others without adult supervision. 

So, JB support interactions can range from amazing to WTF, depending on the developer in charge. In some cases, I find myself dealing with an attitude that treats my inquiry like I am doing this for shits and giggles, and if they shuffle me around with useless questions and boilerplate referrals to company policies, then I will go away. However, when there is an issue reported by users in my plugin, suddenly it is very serious, and they act like it is all on my end.

If you are curious, here are some interactions trying to figure out how to resolve some of the deprecations, and other issues raised by the plugin analyzer.

[ProjectPlainTextFileTypeManager.getFiles is deprecated, replacement PersistentFileSetManager.getFiles is package private](https://intellij-support.jetbrains.com/hc/en-us/community/posts/11323711731346-ProjectPlainTextFileTypeManager-getFiles-is-deprecated-replacement-PersistentFileSetManager-getFiles-is-package-private)

[What does JetBrains prefer to see from Plugin Verifier Deprecated or Experimental](https://intellij-support.jetbrains.com/hc/en-us/community/posts/11397862214290-What-does-JetBrains-prefer-to-see-from-Plugin-Verifier-Deprecated-or-Experimental-?page=1)

[How many years does an API need in purgatory of EXPERIMENTAL status before it is forgiven its sins and considered STABLE](https://intellij-support.jetbrains.com/hc/en-us/community/posts/11398505179666-How-many-years-does-an-API-need-in-purgatory-of-EXPERIMENTAL-status-before-it-is-forgiven-its-sins-and-considered-STABLE-?page=1#community_comment_11399166059410)

**Comment by vsch** [comment]
Unlicensed release uploaded to JB Marketplace. May take a bit for it to funnel through their approval/moderation process.

However, here is a direct link to the download you should be able to use to install plugin from disk: <https://plugins.jetbrains.com/files/7896/331235/MarkdownNavigatorEnhanced-3.0.213.150.zip>

**Comment by ogradyjd** [comment]
Please let me know where we can "donate" for your work.

**Comment by razlani** [comment]
Hey @vsch, I came across this thread as I was on struggle street recently from using the editors built-in plugin, resorting to use remark/unified libs, and came across this open thread.

I scanned the thread and it looks like there's a lot of blood/sweat/tears involved in this project - very upsetting.

I would like to offer some support for what it's worth - what's the current status of the plugin, since you stated incompatibility with the built-in MD? 

If we direct install from disk as above, will that be problematic for future editor updates? 

Finally how can we support you? I'm looking to write a blog and the blogger experience is massively improved with good MD support. 

All the best
razlani

> Unlicensed release uploaded to JB Marketplace. May take a bit for it to funnel through their approval/moderation process.
> 
> However, here is a direct link to the download you should be able to use to install plugin from disk: https://plugins.jetbrains.com/files/7896/331235/MarkdownNavigatorEnhanced-3.0.213.150.zip
